portfolio/src/main/webapp/slideshow.js

```diff
@@ -80,7 +66,9 @@ class SlideShow {
     if(this.mode_ == Mode.MANUAL) {
       this.slides_[this.slideIndex_].style.display = 'none';
       this.slideIndex_ += adjustIndex;
-      this.slideIndex_ %= this.slides_.length;
+      //Fixes modulo division to account for negatives properly
```

Comment:
Space between // and first word.

Also, rephrase this comment, its awkward. Just say "[((a%b) + b) % b] guarantees a is always [0, b) even when a < 0."